### ISSUE TITLE
MYSQL_OPT_RECONNECT is deprecated 

### ISSUE THREAD

**Issue by lionslair** [webmin/webmin#1984]
I have started receiving this error as of an hour ago. Every five minutes from the status monitor

Cron <root@llserver> /etc/webmin/status/monitor.pl

```
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
```

I assume it's generated from the MySQL status check.

Webmin version: 2.101 
MariaDB version 10.8.8 
Perl version v5.30.0

**Comment by lionslair** [comment]
maybe related

https://github.com/perl5-dbi/DBD-mysql/issues/354

https://forum.virtualmin.com/t/mysql-opt-reconnect-is-deprecated/122168/4

**Comment by jcameron** [comment]
Looks like this is due to either a change in the MySQL server or the client libraries, according to https://stackoverflow.com/questions/76875569/warning-mysql-opt-reconnect-is-deprecated-and-will-be-removed-in-a-future-versi

I think the fix has to come from the perl-DBD-mysql package, not Virtualmin.

**Comment by lionslair** [comment]
Yeah it looks that way. Only place I see it reported is webmin status monitor.

I deleted _(because I can not disable)_ the mysql monitor and stopped getting the notification.

Does that mean it is not to do with database connection but how the Mysql host is being checked?

Is there a different way webmin can check the mysql host is up?

**Comment by jcameron** [comment]
You could use the Bootup Action monitor type to check if MySQL is running locally or not..

**Comment by PitWenkin** [comment]
> Does that mean it is not to do with database connection but how the Mysql host is being checked?

Yeah … it's just a warning that one option used in the call to check if the SQL-server is running will be removed in the future.


> [perl5-dbi/DBD-mysql#354](https://github.com/perl5-dbi/DBD-mysql/issues/354)

This seem to be the package that needs to fix this as @jcameron pointed out …

**Comment by PitWenkin** [comment]
What does the monitor.pl actually do?  Only sends email in case something (what exactly?)  is down or does it also do other stuff?

I'm monitoring most services by other means anyway, and am now wondering if I could disbale it.

**Comment by Arthmoor** [comment]
I'm seeing the same messages only from the SQL backup function for the databases I run daily backups off of.

**Comment by jcameron** [comment]
> What does the monitor.pl actually do? Only sends email in case something (what exactly?) is down or does it also do other stuff?

Yes, that's all it does (unless you also have it setup to run other commands if a service is down)

**Comment by lionslair** [comment]
> I'm seeing the same messages only from the SQL backup function for the databases I run daily backups off of.

Same. Will be for the same reason is my guess.

**Comment by PitWenkin** [comment]
> > I'm seeing the same messages only from the SQL backup function for the databases I run daily backups off of.
> 
> Same. Will be for the same reason is my guess.

The backups done via virtualmin have no problems (additional output) … strange

**Comment by Arthmoor** [comment]
The backups do get made, yes, but the system emails a notice for each database that's being backed up, with 2 of the SQL messages as the message body for each one.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.

**Comment by Arthmoor** [comment]
It appears as though the referenced fix for Ubuntu corrected the issue. I have not seen this error in a very long time.

**Comment by iliaross** [comment]
Great! If this issue comes up again or you have a new one, just open a new ticket. Thanks!
